dbt_meshify/utilities/references.py

```diff
@@ -0,0 +1,223 @@
+import re
+from typing import Union
+
+from dbt.contracts.graph.nodes import CompiledNode
+from loguru import logger
+
+from dbt_meshify.change import ChangeSet, EntityType, FileChange, Operation
+from dbt_meshify.dbt_projects import DbtProject, DbtSubProject, PathedProject
+
+
+class ReferenceUpdater:
+    def __init__(self, project: Union[DbtSubProject, DbtProject]):
+        self.project = project
+
+        self.ref_update_methods = {
+            "sql": self.update_refs__sql,
+            "python": self.update_refs__python,
+        }
+        self.source_update_methods = {
+            "sql": self.replace_source_with_ref__sql,
+            "python": self.replace_source_with_ref__python,
+        }
+
+    @staticmethod
+    def update_refs__sql(model_name, project_name, model_code):
```

Comment:
note to future selves: will need to handle optional version args in refs before too long!